mgmtsystem/azure.py

```diff
@@ -440,6 +440,42 @@ def remove_blob_image(self, vm_vhd, container=None):
                        storage_key=self.storage_key,
                        container=container), True)
 
+    def remove_nics_by_search(self, nic_template):
+        """
+        Used for clean_up jobs to remove NIC that are not attached to any test VM
+        """
+        self.logger.info(" Removing NICs with \"{}\" name template".format(nic_template))
+        try:
+            self.run_script(
+                """
+                Get-AzureRmNetworkInterface -ResourceGroupName Automation|
+                Where-Object {{$_.VirtualMachineText -eq "null" -and $_.Name -like "{}"}}|
+                Remove-AzureRmNetworkInterface -Force
+                """.format(nic_template)
+            )
+            return True
+
+        except ActionTimedOutError:
+            return False
+
+    def remove_pips_by_search(self, pip_template):
+        """
+        Used for clean_up jobs to remove public IPs that are not associated to any NIC
+        """
+        self.logger.info(" Removing Public IPs with \"{}\" name template".format(pip_template))
+        try:
+            self.run_script(
+                """
+                Get-AzureRmPublicIpAddress -ResourceGroupName Automation|
+                Where-Object {{$_.Name -like "{}" -and $_.IpConfigurationText -eq "null"}}|
+                Remove-AzureRmPublicIpAddress -Force
+                """.format(pip_template)
+            )
+            return True
```

Comment:
**Required**
I think we need a record of what was found and removed by force.

Consider methods to return the unused resources, and an optional parameter to delete them.  Or leave as is, but at least capture a list of the resources that are being deleted and return it.